### PR TITLE
feat: Add training project for server events

### PR DESCRIPTION
**Pull request by czthiele**
This PR adds a sample project for the PiWeb Server events feature.
Thanks @nicoweh for a lot of reusable code.

### CODE REVIEW

File: PiWeb.Api.Training.Events/Events/CharacteristicCreatedEvent.cs [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+/* Carl Zeiss Industrielle Messtechnik GmbH        */
+/* Softwaresystem PiWeb                            */
+/* (c) Carl Zeiss 2024                             */
+/* * * * * * * * * * * * * * * * * * * * * * * * * */
+
+#endregion
+
+using System;
+
+namespace PiWeb.Api.Training.Events.Events;
```
Comment (nicoweh) [inline PiWeb.Api.Training.Events/Events/CharacteristicCreatedEvent.cs]: Was it intended to change the namespace? How does it work to find the event class when the namespaces are different?

File: PiWeb.Api.Training.Events/Program.cs [hunk 2]
```diff
@@ -0,0 +1,137 @@
...
+        }
+        catch (Exception exception)
+        {
+            ConsoleUI.WriteError(exception.ToString());
+            serverUrl = null;
+            return false;
+        }
+    }
+
+
```
Comment (nicoweh) [inline PiWeb.Api.Training.Events/Program.cs]: Too many empty lines. Also after the method.

**Review by nicoweh — APPROVED** [review]

File: PiWeb.Api.Training.Events/Program.cs (hunk 2 above)
Comment (czthiele) [reply PiWeb.Api.Training.Events/Program.cs]: Done

File: PiWeb.Api.Training.Events/Events/CharacteristicCreatedEvent.cs (hunk 1 above)
Comment (czthiele) [reply PiWeb.Api.Training.Events/Events/CharacteristicCreatedEvent.cs]: Change was intended. As discussed, the events are matched based on method names of the sender hub.

**Pull request merged**